libsolidity/ast/Types.cpp

```diff
@@ -596,17 +595,24 @@ TypePointer FixedPointType::binaryOperatorResult(Token::Value _operator, TypePoi
 		return commonType;
 	if (Token::isBitOp(_operator) || Token::isBooleanOp(_operator))
 		return TypePointer();
-	if (auto fixType = dynamic_pointer_cast<FixedPointType const>(commonType))
+	if (_operator == Token::Exp)
 	{
-		if (Token::Exp == _operator)
-			return TypePointer();
+		return TypePointer();
+//		if (auto intType = dynamic_pointer_cast<IntegerType const>(commonType))
+//			return intType->isSigned() ? TypePointer() : commonType;
+//		else return dynamic_pointer_cast<FixedPointType const>(commonType)->isSigned() ? TypePointer() : commonType;
 	}
 	else if (auto intType = dynamic_pointer_cast<IntegerType const>(commonType))
 		if (intType->isAddress())
```

Comment:
This does not make sense here, the common type cannot be an address.

Reply:
Yep, will fix.